foundry/mouseInteractionManager.d.ts

```diff
@@ -1,213 +1,257 @@
 /**
  * Handle mouse interaction events for a Canvas object.
- *
  * There are three phases of events: hover, click, and drag
+ * @typeParam O - Canvas object this instance handles events for
+ * @typeParam T - Target object for mouseinteraction events. Generally a `ControlIcon` or `O`.
```

Comment:
Is there a way for us to automatically infer `T`?

Reply:
Maybe. It would be very difficult since `T` is essentially determined by `this.options.target`. I don't know if it would be worth the added complexity.

Reply:
OK, I guess we can leave it for now. Maybe we can revisit this later if there is interest in it.